website/docs/provider_reference.html.markdown

```diff
@@ -0,0 +1,95 @@
+---
+layout: "google"
+page_title: "google provider reference"
+sidebar_current: "docs-google-provider-reference"
+description: |-
+  The Google provider is used to configure your GCP project, location, and creds
+---
+
+# `google` provider reference
+The `google` provider block is used to configure default values for your GCP
+project, location (`zone` and `region`), and add your credentials credentials.
+
+-> You can avoid using a provider block by using environment variables. Every
+field of the `google` provider is optional. If you want to share configs between
+environments and deploy to different projects, try it out!
+
+## Example Usage
+
+```hcl
+provider "google" {
+  credentials = "${file("account.json")}"
+  project     = "my-gce-project-id"
+  region      = "us-central1"
+  zone        = "us-central1-c"
+}
+```
+
+## Configuration Reference
+
+The following keys can be used to configure the provider.
+
+* `credentials` - (Optional) Contents of a file that contains your service
+  account private key in JSON format. You can download your existing
+  [Google Cloud service account file]
+  from the Google Cloud Console, or you can create a new one from the same page.
+
+  Credentials can also be specified using any of the following environment
+  variables (listed in order of precedence):
+
+    * `GOOGLE_CREDENTIALS`
+    * `GOOGLE_CLOUD_KEYFILE_JSON`
+    * `GCLOUD_KEYFILE_JSON`
+
+  The [`GOOGLE_APPLICATION_CREDENTIALS`][adc]
+  environment variable can also contain the path of a file to obtain credentials
+  from.
+
+  If no credentials are specified, the provider will fall back to using the
+  [Google Application Default Credentials][adc].
+  If you are running Terraform from a GCE instance, see [Creating and Enabling
+  Service Accounts for Instances][gce-service-account] for details.
+
+  On your computer, if you have made your identity available as the
+  Application Default Credentials by running [`gcloud auth application-default
+  login`][gcloud adc], the provider will use your identity.
+
+  -> [Service accounts][service accounts] are the recommended way
+  to manage GCP credentials. [GCE metadata] is also acceptable, although can
+  only be used in limited scenarios. The `gcloud` method is not guaranteed to work
+  for all APIs.
+
+* `project` - (Optional) The ID of the project to apply any resources to.  This
+  can also be specified using any of the following environment variables (listed
+  in order of precedence):
+
+    * `GOOGLE_PROJECT`
+    * `GOOGLE_CLOUD_PROJECT`
+    * `GCLOUD_PROJECT`
+    * `CLOUDSDK_CORE_PROJECT`
+
+    -> `GOOGLE_CLOUD_PROJECT` is the recommended environment variable to use if
```

Comment:
is there a reason behind this recommendation or did you just decide it was a good idea to have an opinion?

Reply:
I decided it was a good idea to have an opinion because I do not like looking @  like 4 different options. Some of our 1P [Google] tools that interact with Terraform use that var so I chose it.

Reply:
Bikeshed: could we use `GOOGLE_PROJECT` or `GCLOUD_PROJECT` as the suggested default, as that is mirrored in the other environment variable names, where as `GOOGLE_CLOUD_REGION` and `GOOGLE_CLOUD_ZONE` aren't supported, making it an inconsistent prefix?